components/form/ErrorMessage.tsx

```diff
@@ -0,0 +1,29 @@
+import { ErrorMessage } from '@hookform/error-message';
+import { FieldErrors } from 'react-hook-form';
+
+interface FormErrorMessageProps {
+  name: string;
+  errors: FieldErrors;
+  customMessage?: string;
+}
+
+const FormErrorMessage: React.FC<FormErrorMessageProps> = ({
+  name,
+  errors,
+  customMessage,
+}) => {
+  return (
+    <div className='error-message-container'>
+      <ErrorMessage
+        errors={errors}
+        name={name}
+        render={({ message }) => <p className='error-message'>{message}</p>}
+      />
+      {!errors[name] && customMessage && (
```

Comment:
Thank you so much for the valuable feedback @nichgalzin ! I realize I got a bit carried away with the changes. I’ll clear everything out and start fresh on the issue, keeping the changes small and focused to avoid any unintended effects. I’ll also make sure to update the tests so they run successfully. Thanks again for guiding me on this!